### ISSUE TITLE
INTERNAL Error: Failed to bind column reference

### ISSUE THREAD

**Issue by Bilbottom** [duckdb/duckdb#15128]
### What happens?

Running the query (below) fails with the error message:

> INTERNAL Error: Failed to bind column reference "row_id" [44.0] (bindings: {#[57.0], #[57.1], #[57.2], #[57.3], #[57.4]}) This error signals an assertion failure within DuckDB. This usually occurs due to unexpected conditions or errors in the program's logic.

### To Reproduce

It can probably be reduced some more, but here's a smallish example where I'm encountering the error:

```sql
with

grid as (
    from (values ('ABC'), ('DEF')) as v(data)
    select
        unnest(split(data, '')) as letter,
        row_number() over () as row_id,
        generate_subscripts(split(data, ''), 1) AS col_id,
),

search(row_i, col_i, letter_to_match) as (
    values (0, 0, 'A'), (0, 1, 'B'),
)

from (from grid cross join search) as grid_searches
select exists(
    from grid as grid_to_search
    where 1=1
        and grid_searches.row_id = grid_to_search.row_id + grid_searches.row_i
        and grid_searches.col_id = grid_to_search.col_id + grid_searches.col_i
        and grid_searches.letter_to_match = grid_to_search.letter
)
```

I'm happy to provide an explanation of what this intends to do, if needed

### OS:

Windows

### DuckDB Version:

v1.1.1

### DuckDB Client:

JetBrains JDBC

### Full Name:

Bill Wallis

### Affiliation:

Sainsbury's PLC

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Bilbottom** [comment]
Ah oops, I think this duplicates:

- #13542
- #13810

I'm happy to close this one as a duplicate

**Comment by szarnyasg** [comment]
I don't think this duplicates #13542 - that's fixed on nightly but this issue still reproduces.
